internal/controller/genericprovider_controller.go

```diff
@@ -206,7 +206,7 @@ func (r *GenericProviderReconciler) reconcile(ctx context.Context, provider gene
 
 		if !res.IsZero() || err != nil {
 			// the steps are sequential, so we must be complete before progressing.
-			return res, ignoreCoreProviderWaitError(err)
+			return res, err
```

Comment:
so we are basically reverting the https://github.com/kubernetes-sigs/cluster-api-operator/pull/756/files#r2010321261?

Reply:
No, it's just code refactoring, not rerverting